chapters/operatorsandexpressions.tex

```diff
@@ -1479,6 +1469,21 @@ \subsection{Discrete-Time Expressions}\label{discrete-time-expressions}
 functions do not become active between events.
 \end{nonnormative}
 
+For a scalar equation \lstinline!expr1 = expr2! where neither expression is of base type \lstinline!Real!, both expressions must be discrete-time expressions.
```

Comment:
An additional minor point to restore:
```suggestion
For an equation \lstinline!expr1 = expr2! where neither expression is of base type \lstinline!Real!, both expressions must be discrete-time expressions.
```
The reason is that we don't want to unnecessarily for scalarization.

Reply:
I agree that it is better to say "an equation" instead of "a scalar equation". If I see "a scalar equation" I immediately have a question of what happens to an array equation. So at the very least I would say we need to say that for array equations this rule is applied after they were expanded into scalar equation. But then there is also the point Hans made that it would force scalarization.

Reply:
The point of writing it this way is that we start with the simple case, before turning to record and arrays.  So far, the rule doesn't say anything regarding record or array equations, and there is no suggestion to scalarize.  (To be continued in the conversation below.)

Reply:
This now reads:
```
For a scalar or array equation \lstinline!expr1 = expr2! where neither expression is of base type \lstinline!Real!, both expressions must be discrete-time expressions.
```

OK to resolve?

Reply:
Looks good to me. @HansOlsson?

Reply:
Yes.